components/user-mgt/org.wso2.carbon.identity.user.pre.update.profile.action/src/main/java/org/wso2/carbon/identity/user/pre/update/profile/action/internal/rule/PreUpdateProfileActionRuleEvaluationDataProvider.java

```diff
@@ -0,0 +1,163 @@
+import org.wso2.carbon.identity.core.context.model.Flow;
+import org.wso2.carbon.identity.rule.evaluation.api.exception.RuleEvaluationDataProviderException;
+import org.wso2.carbon.identity.rule.evaluation.api.model.Field;
+import org.wso2.carbon.identity.rule.evaluation.api.model.FieldValue;
+import org.wso2.carbon.identity.rule.evaluation.api.model.FlowContext;
+import org.wso2.carbon.identity.rule.evaluation.api.model.FlowType;
+import org.wso2.carbon.identity.rule.evaluation.api.model.RuleEvaluationContext;
+import org.wso2.carbon.identity.rule.evaluation.api.model.ValueType;
+import org.wso2.carbon.identity.rule.evaluation.api.provider.RuleEvaluationDataProvider;
+import org.wso2.carbon.identity.user.action.api.model.UserActionContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Rule evaluation data provider for pre update profile flow.
+ * This class provides the data required for rule evaluation in pre update profile flow.
+ */
+public class PreUpdateProfileActionRuleEvaluationDataProvider implements RuleEvaluationDataProvider {
+
+    private enum RuleField {
+        FLOW_FIELD("flow"),
+        CLAIM_FIELD("claim");
+
+        final String fieldName;
+
+        RuleField(String fieldName) {
+
+            this.fieldName = fieldName;
+        }
+
+        public String getFieldName() {
+
+            return fieldName;
+        }
+
+        public static RuleField valueOfFieldName(String fieldName) throws RuleEvaluationDataProviderException {
+
+            for (RuleField ruleField : RuleField.values()) {
+                if (ruleField.getFieldName().equals(fieldName)) {
+                    return ruleField;
+                }
+            }
+
+            throw new RuleEvaluationDataProviderException("Unsupported field: " + fieldName);
+        }
+    }
+
+    /**
+     * Supported flows for pre update profile action.
+     */
+    private enum ProfileUpdateFlowType {
+        ADMIN_INITIATED_PROFILE_UPDATE("adminInitiatedProfileUpdate"),
+        APPLICATION_INITIATED_PROFILE_UPDATE("applicationInitiatedProfileUpdate"),
+        USER_INITIATED_PROFILE_UPDATE("userInitiatedProfileUpdate");
+
+        final String flowName;
+
+        ProfileUpdateFlowType(String flowName) {
+
+            this.flowName = flowName;
+        }
+
+        public String getFlowName() {
+
+            return flowName;
+        }
+    }
+
+    @Override
+    public FlowType getSupportedFlowType() {
+
+        return FlowType.PRE_UPDATE_PROFILE;
+    }
+
+    @Override
+    public List<FieldValue> getEvaluationData(RuleEvaluationContext ruleEvaluationContext,
+                                              FlowContext flowContext, String tenantDomain)
+            throws RuleEvaluationDataProviderException {
+
+        List<FieldValue> fieldValueList = new ArrayList<>();
+
+        for (Field field : ruleEvaluationContext.getFields()) {
+            switch (RuleField.valueOfFieldName(field.getName())) {
```

Comment:
Why did we introduce this method ? `RuleField.valueOfFieldName` ?

Can't we simply use field.getName to the switch input ?